src/Adapter/MSTestAdapter.PlatformServices/ObjectModel/UnitTestElement.cs

```diff
@@ -246,7 +241,32 @@ private Guid GenerateSerializedDataStrategyTestId(string testFullName)
             hash.Append(CloseBracket);
         }
 
-        return new Guid(hash.GetCurrentHash());
+        byte[] hashBytes = hash.GetCurrentHash();
+
+        return VersionedGuidFromHash(hashBytes, hashVersion);
+    }
+
+    internal /* for testing */ static Guid VersionedGuidFromHash(byte[] hashBytes, byte hashVersion)
+    {
+        const int firstByte = 0;
```

Comment:
I would go with `3` so that when you do the `ToString()`, it's the first hex character that is representing our own version.

Reply:
this also ends up depending on whether or not we are on big endian system, detailed in code.